File: js/1-9/problem7.js
// sieve of eratoshenes
var primes = function(n) {
	var answer = [];
	var sieve = function() {
		var arr = [];
		for (let i = 0; i < n; i++) {
			arr.push(true);
		}
		return arr;
	}();
	for (let p = 2; p <= n; p++) {
		if (sieve[p]) {
			answer.push(p)
			for (let j = p * p; j <= n + 1; j += p) {
				sieve[j] = false;
			}
		}
	}
	return answer;
};

var place = 10001;

console.log(primes(1000000)[place - 1]);

File: python/1-9/problem7.py
# sieve of eratosthenes
def primes(n):
    ps, sieve = [], [True] * (n + 1)
    for p in range(2, n + 1):
        if sieve[p]:
            ps.append(p)
            for i in range(p * p, n + 1, p):
                sieve[i] = False
    return ps


def nth_prime(n):
    p = primes(n * 100)
    return p[n - 1]


if __name__ == '__main__':
    print(nth_prime(10001))
